test/e2e/util/util.go

```diff
@@ -1549,3 +1552,119 @@ func validateHostedClusterConditions(t *testing.T, ctx context.Context, client c
 	}
 	t.Logf("Successfully validated all expected HostedCluster conditions in %s", duration)
 }
+
+func CheckAffinityOpinions(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("CheckAffinityOpinions", func(t *testing.T) {
+		namespace := manifests.HostedControlPlaneNamespace(hostedCluster.Namespace, hostedCluster.Name).Name
+		awsEbsCsiDriverOperatorPodSubstring := "aws-ebs-csi-driver-operator"
+
+		var podList corev1.PodList
+		if err := client.List(ctx, &podList, crclient.InNamespace(namespace)); err != nil {
+			t.Fatalf("failed to list pods in namespace %s: %v", namespace, err)
+		}
+
+		namespacedName := types.NamespacedName{
+			Namespace: namespace,
+			Name:      hostedCluster.Name,
+		}
+
+		var hcp hyperv1.HostedControlPlane
+		if err := client.Get(ctx, namespacedName, &hcp); err != nil {
+			t.Fatalf("failed to get hostedcontrolplane: %v", err)
+		}
+
+		DC := suppconfig.DeploymentConfig{}
+		DC.SetDefaults(&hcp, nil, k8sutilspointer.Int(1))
```

Comment:
I think we wouldn't want this e2e validation to trust `SetDefaults()`
Instead we'd want to define our raw expectations (e.g. see how we do it in `TestSetLocation()`)

Reply:
If we change any affinities in SetDefaults() and I refactor this code to be like TestSetLocation() then we will have to manually update the test so it doesn't break.  I think we are less likely to break the test using SetDefaults() and it will be easier to add multizoneSpread and requestServing test cases. 

I'm happy to make it like TestSetLocation() for now but  I think it should use SetDefaults() when we're done with the refactoring of that package, wdyt? @enxebre

Reply:
>then we will have to manually update the test so it doesn't break

Right, that's literally what we want and what adds value to this check. Decoupling the expected output from the implementation so we have a clear fail signal if we break it by accident.